planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -2965,7 +2971,22 @@ TurnSignalInfo AvoidanceModule::calcTurnSignalInfo(const ShiftedPath & path) con
       turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_RIGHT;
     }
   } else {
-    turn_signal_info.turn_signal.command = TurnIndicatorsCommand::DISABLE;
+    for (const auto & cl : *debug_data_.current_lanelets) {
+      const auto neigbor_lane = planner_data_->route_handler->getAllSharedLineStringLanelets(cl);
+      for (const auto & lane : neigbor_lane) {
+        LinearRing2d lane_points;
+        for (const auto & p : lane.polygon2d().basicPolygon()) {
+          lane_points.push_back(Point2d(p.x(), p.y()));
+        }
+        if (boost::geometry::intersects(path_points, lane_points)) {
+          if (segment_shift_length > 0.0) {
+            turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_LEFT;
+          } else {
+            turn_signal_info.turn_signal.command = TurnIndicatorsCommand::ENABLE_RIGHT;
+          }
+        }
+      }
+    }
```

Comment:
- I think it better to use the points of shifted_path where in the range of shift segment instead of iterating all the path points.( only the path points between `front_shift_line.start_idx` and `front_shift_line.end_idx`  should be enough ) 

- Instead of using lanelet polygon you can use current lane's right and left bound. (If `segment_shift_length > 0.0 `then get left bound and if `segment_shift_length < 0.0` get right bound )

- It is not realistic to use path points for intersection because what matter in this scenario is that vehicle polygon is exceeding the current lane or not (means occupying any neighbor lane). So you should use vehicle dimensions and transform ego polygon to related path points

- Please add some comment lines which explain your purpose

Reply:
> Instead of using lanelet polygon you can use current lane's right and left bound. (If segment_shift_length > 0.0 then get left bound and if segment_shift_length < 0.0 get right bound )

I think this is simple and good solution to achieve what you do, too.

Reply:
@beyzanurkaya please make sure that you actually applied the necessary changes before you resolve the conversation. :smile: